README.md

```diff
@@ -1,25 +1,40 @@
 # IETF Multiformats Working Group
 
 Multiformats are a collection of self-describing data formats that consist of an
-inline data header and a data value. Multiformats have binary and text-based
-representations and are used to express base encodings (multibase),
-cryptographic hashes (multihash), cryptographic keys (multikey), network
-addresses (multiaddr), and a variety of other binary serialization formats
-(multicodec).
+inline data header and a data value. They are composable in that some of the
+more complex forms take another multiformat header/value tuple as value,
+recursively. Multiformats are designed to compose into binary streams with both
+header values and data values expressed in unsigned varints (a slight variation
+from protobuf varints and the UNIX LEB128 standard), with the exception of
```

Comment:
it would be nice to add references for this, and clarify why neither is acceptable  for the use case.

Reply:
@OR13 , fair points all around.

Upon further research I have actually learned that `unsigned varints` are, in fact, neither mandatory nor novel-- it's actually just the LEB128 [defined in 1993 by the DWARF standard](http://dwarfstd.org/doc/dwarf-2.0.0.pdf) but "minimally expressed".  And while the end-user of the IPFS stack sees most binary values expressed in that form by IPFS tooling, it may open up more usecases to consider that an implementation detail and use "raw binary" bytes as the unique key in the multicodecs table going forward.  For the relationship between LEB128 and the varints used in IPFS tooling see [this earlier draft of a multihash spec from 2020](http://dwarfstd.org/doc/dwarf-2.0.0.pdf).

Reply:
incorporated the above into the multibase input doc and updated this proposal to not mention the rest of the multi-fam.  i think the goal is for each spec+registry at IETF to be usable on its own (except where it depends on another IETF spec+registry), so cleaner to have this WG work on exactly two multis.

Reply:
> so cleaner to have this WG work on exactly two multis.

this will create a registry table inconsistency potentially... in case the registry changes an entry for something that is not a hash... I suppose it would be up to the designated experts to decide if that is ok.

Reply:
you mean inconsistency between the older community registry and the more authoritative IETF one? well the good news is the two designated experts (at least for the foreseeable future) are already the managers of that other registry, so at least they would see the inconsistencies coming if anyone tried to register a multibase or a multihash over any other multi* entry in the original registry.

Reply:
for example:

- hashes are here: https://github.com/multiformats/multicodec/blob/master/table.csv#L12
- keys are here: https://github.com/multiformats/multicodec/blob/master/table.csv#L174

If the IANA / IETF registry disagrees with this github repo, I would expect problems... experts would be required to prevent this disagreement from happening... that essentially means "pre-reserving" the code point space... for anything that is not a hash.

Reply:
would you recommend entering `CIDv1`, `CIDv2`, and all "draft" or "final" status registries in the multicodecs from the other multi specs as `reserved`, to pre-empt that possible conflict?

Reply:
the community registry has a lot of unimplemented cruft in it and stuff, so the good news is that it won't be 400 mysterious reservations, but even limiting it to what's currently implemented and in prod use, demoting a few `final`s to `draft`s (for lack of multiple testable implementations) and `draft`s to `experimental`s, it will still be a lot more code points than are used by multihash!

Reply:
if the entire identifier space is changing, this is not an issue... it only impacts registering existing identifiers.